src/commonMain/kotlin/fr/acinq/lightning/crypto/KeyManager.kt

```diff
@@ -29,15 +30,21 @@ interface KeyManager {
 
     fun commitmentPoint(channelKeyPath: KeyPath, index: Long): PublicKey
 
-    fun channelKeyPath(localParams: LocalParams, channelVersion: ChannelVersion): KeyPath =
+    fun commitmentSecret(shaSeed: ByteVector32, index: Long): PrivateKey
+
+    fun commitmentPoint(shaSeed: ByteVector32, index: Long): PublicKey
+
+    fun channelKeyPath(fundingKeyPath: KeyPath, channelVersion: ChannelVersion): KeyPath =
         if (channelVersion.isSet(ChannelVersion.USE_PUBKEY_KEYPATH_BIT)) {
             // deterministic mode: use the funding pubkey to compute the channel key path
-            channelKeyPath(fundingPublicKey(localParams.fundingKeyPath))
+            channelKeyPath(fundingPublicKey(fundingKeyPath))
         } else {
             // legacy mode:  we reuse the funding key path as our channel key path
-            localParams.fundingKeyPath
+            fundingKeyPath
         }
 
+    fun channelKeyPath(localParams: LocalParams, channelVersion: ChannelVersion): KeyPath = channelKeyPath(localParams.channelKeys.fundingKeyPath, channelVersion)
```

Comment:
Do we want to keep that function exposed?
I think we can make the interface of the `KeyManager` is bit smaller to avoid mis-using it.

Reply:
I'd like to keep it for now, it feels a bit cleaner to use `localParams`